### DISCUSSION TITLE
I can't find the correct type for query

### DISCUSSION THREAD

**Discussion by hugocruzlfc** [mikro-orm/mikro-orm#5432] [General]
### Describe the bug

I just upgraded to version (6.1) and I get this error:

Error: The module '"@mikro-orm/core"' does not have any exported 'Query' members.

![Captura de pantalla 2024-04-09 a las 13 51 20](https://github.com/mikro-orm/mikro-orm/assets/55498983/11d3b2b3-000e-4d17-980b-bcb9071e787b)

### Reproduction

I use this type to have correctly the type of my filters:

const queryFilters: Array<Query<IncidenceEntity>> = [];

### What driver are you using?

@mikro-orm/postgresql

### MikroORM version

├─┬ @mikro-orm/cli@6.1.12 │ ├── @mikro-orm/core@6.1.12 deduped │ └─┬ @mikro-orm/knex@6.1.12 │   └── @mikro-orm/core@6.1.12 deduped ├── @mikro-orm/core@6.1.12 ├─┬ @mikro-orm/entity-generator@6.1.12 │ └── @mikro-orm/core@6.1.12 deduped ├─┬ @mikro-orm/migrations@6.1.12 │ └── @mikro-orm/core@6.1.12 deduped ├─┬ @mikro-orm/nestjs@5.2.3 │ └── @mikro-orm/core@6.1.12 deduped ├─┬ @mikro-orm/postgresql@6.1.12 │ └── @mikro-orm/core@6.1.12 deduped └─┬ @mikro-orm/reflection@6.1.12   └── @mikro-orm/core@6.1.12 deduped

### Node.js version

v18.19.0 ├─┬ @commitlint/cli@18.4.3 │ └─┬ @commitlint/load@18.4.3 │   ├─┬ cosmiconfig-typescript-loader@5.0.0 │   │ └── typescript@5.4.4 deduped │   └─┬ cosmiconfig@8.3.6 │     └── typescript@5.4.4 deduped ├─┬ @nestjs/cli@10.3.2 │ ├─┬ fork-ts-checker-webpack-plugin@9.0.2 │ │ └── typescript@5.4.4 deduped │ └── typescript@5.3.3 ├─┬ @nestjs/schematics@10.1.1 │ └── typescript@5.4.4 deduped ├─┬ @typescript-eslint/eslint-plugin@6.13.2 │ └─┬ ts-api-utils@1.0.2 │   └── typescript@5.4.4 deduped ├─┬ eslint-config-standard-with-typescript@39.1.1 │ └── typescript@5.4.4 deduped ├─┬ ts-jest@29.1.2 │ └── typescript@5.4.4 deduped ├─┬ ts-loader@9.5.1 │ └── typescript@5.4.4 deduped ├─┬ ts-node@10.9.2 │ └── typescript@5.4.4 deduped └── typescript@5.4.4

### Operating system

macOs

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Slack](https://join.slack.com/t/mikroorm/shared_invite/enQtNTM1ODYzMzM4MDk3LWM4ZDExMjU5ZDhmNjA2MmM3MWMwZmExNjhhNDdiYTMwNWM0MGY5ZTE3ZjkyZTMzOWExNDgyYmMzNDE1NDI5NjA).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
`Query` is a rather internal type, you should be using `FilterQuery` or `ObjectQuery` (where `FilterQuery` also accepts PK or array of PKs, `ObjectQuery` is always an object).

edit: also, the `Query` type was renamed to `ExpandQuery` some time ago (and exported, up until that time it was internal type that wasn't even part of public exports, you were never supposed to be using that really)

**Reply by hugocruzlfc** [reply]
Thanks for the prompt reply, we are upgrading our project from v5.9 => v6.1 and we are experiencing some problems with these types-
